### PR TITLE
Change noisy logs on SegRep and remote store paths to trace level

### PR DESCRIPTION
**Pull request by mch2**
### Description
Change noisy logs on SegRep and remote store paths to trace level

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/8452

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dreamer-89 — APPROVED** [review]

File: server/src/main/java/org/opensearch/index/shard/IndexShard.java [hunk 1]
```diff
@@ -1664,7 +1664,7 @@ public boolean isSegmentReplicationAllowed() {
             return false;
         }
         if (getReplicationEngine().isEmpty()) {
-            logger.warn(
+            logger.trace(
```
Comment (dreamer-89) [inline server/src/main/java/org/opensearch/index/shard/IndexShard.java:1667]: nit: Is it possible to get into this state i.e. where engine is not defined or it is not `NRTReplicationEngine` ?

**Comment by mch2** [comment]
https://github.com/opensearch-project/OpenSearch/issues/7679

**Comment by mch2** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/22710/
> * **CommitID:** [b47f182](https://github.com/opensearch-project/OpenSearch/commit/b47f182343108ea6d71ca9c4dbd714158f3f694b)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/9326

File: server/src/main/java/org/opensearch/index/shard/IndexShard.java (hunk 1 above)
Comment (dreamer-89) [reply server/src/main/java/org/opensearch/index/shard/IndexShard.java:1667]: Thanks @mch2. Yes, we can take this in a separate change. Please feel free to add a `todo` comment or create a tracking issue.

**Comment by dreamer-89** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/22713/
> * **CommitID:** [b47f182](https://github.com/opensearch-project/OpenSearch/commit/b47f182343108ea6d71ca9c4dbd714158f3f694b)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

More flakies :(

Screenshot 2023-08-14 at 4 47 58 PM

**Review by Poojita-Raj — APPROVED** [review]

**Comment by dreamer-89** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** UNSTABLE ❕
> * **TEST FAILURES:**
> 
> ```
>       1 org.opensearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT.test {yaml=pit/10_basic/Delete all}
>       1 org.opensearch.remotestore.SegmentReplicationUsingRemoteStoreIT.testDropPrimaryDuringReplication
>       1 org.opensearch.action.admin.cluster.node.tasks.ResourceAwareTasksTests.testTaskResourceTrackingDuringTaskCancellation
> ```
> 
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/22717/
> * **CommitID:** [b47f182](https://github.com/opensearch-project/OpenSearch/commit/b47f182343108ea6d71ca9c4dbd714158f3f694b)
>   Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

testDropPrimaryDuringReplication -> https://github.com/opensearch-project/OpenSearch/issues/8059
testTaskResourceTrackingDuringTaskCancellation -> https://github.com/opensearch-project/OpenSearch/issues/5329
SmokeTestMultiNodeClientYamlTestSuiteIT.test {yaml=pit/10_basic/Delete all} -> https://github.com/opensearch-project/OpenSearch/issues/9332

**Pull request merged**